apiserver/clusterservice/scaleimpl.go

```diff
@@ -292,49 +334,179 @@ func ScaleDown(deleteData bool, clusterName, replicaName, ns string) msgs.ScaleD
+	} else if shutdown {
+
+		log.Debugf("no replicas remain for cluster %s, scaling primary deployment %s to 0",
+			clusterName, primaryDeploymentName)
+
+		clusterServices, err := scaleClusterDeployments(cluster, 0)
+		if err != nil {
+			response.Status.Code = msgs.Error
 			response.Status.Msg = err.Error()
+			return response
+		}
+
+		// set the status of the pgcluster to 'pgcluster Shutdown' with associated message
+		message := "Cluster shutdown by scaling down the cluster"
+		if err := kubeapi.PatchpgclusterStatus(apiserver.RESTClient, crv1.PgclusterStateShutdown,
+			message, &cluster, ns); err != nil {
+			response.Status.Code = msgs.Error
+			response.Status.Msg = err.Error()
+			return response
+		}
+
+		if err := publishClusterShutdown(&cluster); err != nil {
+			log.Error(err)
 		}
+
+		response.Results = append(response.Results, fmt.Sprintf("The primary server is shutdown "+
+			"and the database for cluster %s has been stopped.\nThe following services have also "+
+			"been disabled: %s\nScale up the cluster to restart the database and re-enable any "+
+			"supporting services.", clusterName, strings.Join(clusterServices, ", ")))
+
 		return response
 	}
 
+	// check to see if the replica name provided matches the name of the primary or any of the
+	// replicas found for the cluster
+	var replicaNameFound bool
+	for _, pod := range replicaList.Items {
+		if pod.Labels[config.LABEL_DEPLOYMENT_NAME] == replicaName {
+			replicaNameFound = true
+			break
+		}
+	}
+	// return an error if the replica name provided does not match the primary or any replicas
+	if !replicaNameFound {
+		response.Status.Code = msgs.Error
+		response.Status.Msg = fmt.Sprintf("Unable to find replica with name %s",
+			replicaName)
+	}
+
 	//create the rmdata task which does the cleanup
 
+	clusterPGHAScope := cluster.ObjectMeta.Labels[config.LABEL_PGHA_SCOPE]
 	deleteBackups := false
 	isReplica := true
 	isBackup := false
 	taskName := replicaName + "-rmdata"
-	clusterPGHAScope := cluster.ObjectMeta.Labels[config.LABEL_PGHA_SCOPE]
 	err = apiserver.CreateRMDataTask(clusterName, replicaName, taskName, deleteBackups, deleteData, isReplica, isBackup, ns, clusterPGHAScope)
 	if err != nil {
 		response.Status.Code = msgs.Error
 		response.Status.Msg = err.Error()
 		return response
 	}
 
-	response.Results = append(response.Results, "deleted Pgreplica "+replicaName)
+	response.Results = append(response.Results, "deleted replica "+replicaName)
 	return response
 }
+
+// scaleClusterDeployments scales all deployments for a cluster to the number of replicas
+// specified using the 'replicas' parameter.  This is typically used to scale-up or down the
+// primary deployment and any supporting services (pgBackRest and pgBouncer) when shutting down
+// or starting up the cluster due to a scale or scale-down request.
+func scaleClusterDeployments(cluster crv1.Pgcluster, replicas int) (clusterServices []string,
+	err error) {
+
+	clusterName := cluster.Name
```

Comment:
do we want `cluster.Spec.ClusterName` here, given `cluster.Name` changes based on the name of the Deployment? I recall it's one of those.

Reply:
Yup it's `cluster.Spec.ClusterName` -- see the operator/cluster/pgbouncer.go for some examples :smile:

Reply:
Not tracking here.  The cluster name should not be updated following a restore.  The name found under `cluster.Name` is the name included in the `ObjectMeta` for the CR, and cannot be changed once the CR has been created.

Multiple names are definitely redundant, however.  I'm thinking the name should be removed from `Spec` altogether, and we should just leverage the name provided via the `ObjectMeta`.

Reply:
I may be mixing up cluster.Spec.Name, cluster.Name and cluster.Spec.ClusterName :|

Reply:
Per our conversation on 3/16, `cluster.Name` can be utilized to obtain the proper cluster name.  Therefore, no additional changes will be made at this time.